CMakeLists.txt

```diff
@@ -95,8 +96,8 @@ if (UNIT_TESTING)
   # make generate step fail if cmocka dependency is not found
   find_package(cmocka CONFIG REQUIRED)
 
-  # When using VCPKG, cmocka lib name is set different than when it is globally installed
-  if(DEFINED ENV{VCPKG_ROOT} OR DEFINED ENV{VCPKG_INSTALLATION_ROOT})
+  # Old versions of cmocka (including the latest stable) don't define an imported target
+  if(NOT TARGET cmocka::cmocka)
```

Comment:
I don't understand this change. Can you please explain it a bit more.

Reply:
So the last cmocka release was in like 2019, this release includes a cmocka-config.cmake file but it does not include any imported targets in that config file. Sometime since then cmocka has been updated to include such targets.

Because of this when cmocka is "globally installed" and the distro package happens to just be latest-stable this would break. Additionally, cmocka has since been updated in vcpkg to be a snapshot that includes the targets. So the old check was wrong both for many instances of using vcpkg and for many instances of not using it.

The new check just directly checks if the target is provided, that way we'll start using the target as soon as it appears no matter what, and things continue to work on the old version of vcpkg used in CI.

(actually, this whole check was non-functional before because CMOCKA_LIBRARIES was reference in the tests instead of CMOCKA_LIB, I fixed that too)